packages/app/src/debug/utils/DebugMapping.ts

```diff
@@ -1,27 +1,64 @@
-export const specsList = <S extends {id: string}, T extends {specId: string}>(specs: readonly S[], tests: readonly T[]): {spec: S, tests: T[]}[] => {
-  const mappedTests = tests.reduce((acc, curr) => {
-    // console.log(`test specId = ${ curr.specId}`)
-    let spec = acc[curr.specId]
+import type { DebugSpecListSpecFragment, DebugSpecListTestsFragment, DebugLocalSpecsFragment, TestingTypeEnum, DebugSpecListGroupsFragment } from '../../generated/graphql'
+import { posixify } from '../../paths'
 
-    if (!spec) {
+type DebugSpecsArgs = {
+  specs: readonly DebugSpecListSpecFragment[]
+  tests: readonly DebugSpecListTestsFragment[]
+  groups: readonly DebugSpecListGroupsFragment[]
+  localSpecs: readonly DebugLocalSpecsFragment[]
+  currentTestingType: TestingTypeEnum
+}
+
+export type CloudDebugSpec = {
+  spec: DebugSpecListSpecFragment
+  tests: DebugSpecListTestsFragment[]
+  groups: DebugSpecListGroupsFragment[]
+  foundLocally: boolean
+  testingType: TestingTypeEnum
+  matchesCurrentTestingType: boolean
+}
+
+export const specsList = ({ specs, tests, localSpecs, currentTestingType, groups }: DebugSpecsArgs): CloudDebugSpec[] => {
+  const localSpecsSet = new Set(localSpecs.map(((spec) => posixify(spec.relative))))
+  const groupsMap = groups.reduce<{[id: string]: DebugSpecListGroupsFragment}>((acc, group) => ({ ...acc, [group.id]: group }), {})
+
+  const mappedTests = tests.reduce<{[id: string]: CloudDebugSpec}>((acc, curr) => {
+    let debugResult = acc[curr.specId]
+
+    if (!debugResult) {
       const foundSpec = specs.find((spec) => spec.id === curr.specId)
 
-      spec = { spec: foundSpec }
-      // console.log('looking for spec', spec)
```

Comment:
Remove this comment?

Reply:
I removed this in a previous commit, maybe the PR was stale when you were reviewing?